src/app/projects/edit-project/edit-project.component.ts

```diff
@@ -0,0 +1,65 @@
+import { ChangeDetectorRef, Component, Inject, OnInit } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup } from "@angular/forms";
+import { ProjectsService } from "../projects.service";
+import { MAT_BOTTOM_SHEET_DATA, MatBottomSheetRef } from "@angular/material";
+import { Project } from "../project";
+import { HttpErrorResponse } from "@angular/common/http";
+import { GlobalErrorHandlerService } from "../../error/global-error-handler.service";
+
+@Component({
+  selector: 'app-edit-project',
+  templateUrl: './edit-project.component.html',
+  styleUrls: ['./edit-project.component.scss']
+})
+export class EditProjectComponent implements OnInit {
+
+  projectForm: FormGroup;
+  errorMessage: string = null;
+
+  constructor(@Inject(MAT_BOTTOM_SHEET_DATA) public project: Project,
+              private projectService: ProjectsService,
+              private formBuilder: FormBuilder,
+              private bottomSheet: MatBottomSheetRef,
+              private changeDetector: ChangeDetectorRef,
+              private globalErrorHandlerService: GlobalErrorHandlerService) {
+    this.projectForm = formBuilder.group({
+      name: new FormControl(project.name),
+      customer: new FormControl(project.customer),
+      industrySector: new FormControl(project.industrySector),
+      description: new FormControl(project.description)
+    });
+  }
+
+  ngOnInit() {
+  }
+
+  editProject() {
+    if (!this.projectForm.valid) {
```

Comment:
I am not sure if we need such a check here. From my point of view it is the UI / template who is responsible for preventing a user from updating of a project when the form is invalid. So I would not duplicate the check and remove it from here. But it is up to you.

Reply:
done